lib/Serialization/Serialization.cpp

```diff
@@ -4774,6 +4775,22 @@ void swift::serializeToBuffers(
       *moduleDocBuffer = llvm::make_unique<llvm::SmallVectorMemoryBuffer>(
                            std::move(buf), options.DocOutputPath);
   }
+
+  if (!StringRef::withNullAsEmpty(options.SourceInfoOutputPath).empty()) {
+    SharedTimer timer("Serialization, swiftsourceinfo, to buffer");
+    llvm::SmallString<1024> buf;
+    llvm::raw_svector_ostream stream(buf);
+    writeSourceInfoToStream(stream, DC);
```

Comment:
@harlanhaskins This always using a buffer is unfortunate; ideally we'd only use it when we're going to keep it in memory.